core-cube/src/main/java/org/apache/kylin/cube/CubeInstance.java

```diff
@@ -717,4 +724,51 @@ public static CubeSegment findSegmentWithJobId(String jobID, CubeInstance cubeIn
         throw new IllegalStateException("No segment's last build job ID equals " + jobID);
     }
 
+    public CubeInstance findLastSnapshot(List<RealizationEntry> realizationEntries, String lookupTableName) {
+        if (realizationEntries.size() > 0) {
+            List<IRealization> realizationList = Lists.newArrayList();
+            RealizationRegistry registry = RealizationRegistry.getInstance(config);
+            for (int i = 0; i < realizationEntries.size(); i++) {
+                IRealization realization = registry.getRealization(realizationEntries.get(i).getType(), realizationEntries.get(i).getRealization());
+                if (realization == null) {
+                    logger.error("Realization '" + realizationEntries.get(i) + " is not found '" + this.getName() + "'");
+                    continue;
+                }
+                if (realization.isReady() == false) {
+                    logger.error("Realization '" + realization.getName() + " is disabled '" + this.getName() + "'");
+                    continue;
+                }
+                if (realization.getModel().isLookupTable(lookupTableName)) {
+                    realizationList.add(realization);
+                }
+            }
+            this.realizations = realizationList.toArray(new IRealization[realizationList.size()]);
+            List<CubeInstance> allcubesList = new ArrayList<CubeInstance>();
+            for (int i = 0; i < realizations.length; i++) {
+                if (realizations[i] instanceof CubeInstance){
+                    allcubesList.add((CubeInstance) realizations[i]);
+                }
+            }
+            Map<CubeInstance, Long> map = new HashMap();
+            for (CubeInstance cube : allcubesList) {
```

Comment:
To find the latest built cube among the project's realizations, this can be done in one loop. Now it is in 3 loops.